src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,230 @@
+    peopleList: Person[],
+    centuries: string[],
+  ): Person[] => {
+    if (centuries.length === 0) {
+      return peopleList;
+    }
+
+    return peopleList.filter(person => {
+      const birthYear = person.born;
+
+      const personCentury = birthYear
+        ? Math.floor((birthYear - 1) / 100) + 1
+        : null;
+
+      return personCentury && centuries.includes(String(personCentury));
+    });
+  };
+
+  const filterPeople = (
+    peopleList: Person[],
+    query: string,
+    sex: string | null,
+    centuries: string[],
+  ): Person[] => {
+    let filteredPeople = filterPeopleByQuery(peopleList, query);
+
+    filteredPeople = filterPeopleBySex(filteredPeople, sex);
+    filteredPeople = filterPeopleByCentury(filteredPeople, centuries);
+
+    return filteredPeople;
+  };
+
+  const handleSort = (field: string) => {
+    let newOrder: SortState['order'] = 'asc';
+
+    if (sortState.field === field) {
+      if (sortState.order === 'asc') {
+        newOrder = 'desc';
+      } else if (sortState.order === 'desc') {
+        newOrder = null;
+      }
+    }
+
+    setSortState({ field: newOrder ? field : null, order: newOrder });
+
+    const params = new URLSearchParams(searchParams);
+
+    if (newOrder) {
+      params.set('sort', field);
+
+      if (newOrder === 'desc') {
+        params.set('order', newOrder);
+      }
+    } else {
+      params.delete('sort');
+      params.delete('order');
+    }
+
+    setSearchParams(params);
+  };
+
+  const sortPeople = (peopleList: Person[]): Person[] => {
+    if (!sortState.field || !sortState.order) {
+      return peopleList;
+    }
+
+    const sortedPeople = [...peopleList].sort((a, b) => {
+      const aValue = a[sortState.field as keyof Person]!;
+      const bValue = b[sortState.field as keyof Person]!;
+
+      if (aValue < bValue) {
+        return sortState.order === 'asc' ? -1 : 1;
+      }
+
+      if (aValue > bValue) {
+        return sortState.order === 'asc' ? 1 : -1;
+      }
+
+      return 0;
+    });
+
+    return sortedPeople;
+  };
+
+  const renderSortIcon = (field: string) => {
+    if (sortState.field !== field) {
+      return <i className="fas fa-sort"></i>; // Іконка за замовчуванням
+    }
+
+    if (sortState.order === 'asc') {
+      return <i className="fas fa-sort-up"></i>; // Стрілка вгору
+    }
+
+    if (sortState.order === 'desc') {
+      return <i className="fas fa-sort-down"></i>; // Стрілка вниз
+    }
+
+    return <i className="fas fa-sort"></i>; // Іконка за замовчуванням
+  };
+
+  useEffect(() => {
+    getPeople()
+      .then(data => {
+        let filteredPeople = filterPeople(
+          data,
+          queryParam,
+          sexParam,
+          centuryParam, // Без split, бо це вже масив
+        );
+
+        filteredPeople = sortPeople(filteredPeople);
+
+        setPeople(filteredPeople);
+      })
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  }, [searchParams]);
```

Comment:
The `useEffect` dependency array includes `searchParams`, which is correct for updating the component when search parameters change. However, ensure that `getPeople` is memoized or stable to prevent unnecessary re-renders.